pkg/sdk/validation/manifest/json_remote_implementation.go

```diff
@@ -93,10 +114,82 @@ func (v *RemoteImplementationValidator) Do(ctx context.Context, _ types.Manifest
 }
 
+func (v *RemoteImplementationValidator) validateInputArtifactsNames(ctx context.Context, entity types.Implementation) (ValidationResult, error) {
+	var validationErrs []error
+	var interfacesInputNames []string
+	var implAdditionalInput []string
+	var workflowArtifacts []wfv1.Artifact
+
+	//1. get implementations inputs
+	for _, implementsItem := range entity.Spec.Implements {
+		interfaceInput, err := getTypesForImplementation(ctx, implementsItem.Path, v.hub)
+		if err != nil {
+			return ValidationResult{}, errors.Wrap(err, "while getting types for implementation")
+		}
+		for _, inputParameter := range interfaceInput.Parameters {
+			interfacesInputNames = append(interfacesInputNames, inputParameter.Name)
+		}
+		for _, inputTypeInstance := range interfaceInput.TypeInstances {
+			interfacesInputNames = append(interfacesInputNames, inputTypeInstance.Name)
+		}
+	}
+
+	//2. get inputs from first workflow template
+	workflow, err := createWorkflow(entity.Spec.Action.Args)
+	if err != nil {
+		return ValidationResult{}, errors.Wrap(err, "while creating Workflow object")
+	}
+	if len(workflow.Templates) > 0 {
+		workflowArtifacts = append(workflowArtifacts, workflow.Templates[0].Inputs.Artifacts...)
+	}
+
+	//3. get additional inputs
+	if entity.Spec.AdditionalInput != nil {
+		for name := range entity.Spec.AdditionalInput.Parameters {
+			implAdditionalInput = append(implAdditionalInput, name)
+		}
+		for name := range entity.Spec.AdditionalInput.TypeInstances {
+			implAdditionalInput = append(implAdditionalInput, name)
+		}
+	}
+
+	//4. verified optional parameter on input parameter
+	for _, artifact := range workflowArtifacts {
+		existsInInterface := slices.Contains(interfacesInputNames, artifact.Name)
+		if existsInInterface &&
+			artifact.Optional {
+			validationErrs = append(validationErrs, fmt.Errorf("workflow artifact input '%s' is optional but it is in interface", artifact.Name))
+		}
+		existsInAdditionalInput := slices.Contains(implAdditionalInput, artifact.Name)
+		if existsInAdditionalInput && !artifact.Optional {
+			validationErrs = append(validationErrs, fmt.Errorf("workflow artifact input '%s' is not optional", artifact.Name))
+		}
+		if !existsInInterface && !existsInAdditionalInput {
+			validationErrs = append(validationErrs, fmt.Errorf("workflow artifact input '%s' does not exists neither in interface input nor additional input", artifact.Name))
+		}
+	}
+
+	return ValidationResult{Errors: validationErrs}, nil
+}
+
+func createWorkflow(rawWorkflowSpec map[string]interface{}) (*argo.Workflow, error) {
+	var renderedWorkflow = struct {
+		Spec argo.Workflow `json:"workflow"`
+	}{}
+
+	b, err := json.Marshal(rawWorkflowSpec)
+	if err != nil {
+		return nil, errors.Wrap(err, "while marshaling Implementation workflow")
+	}
+	if err := json.Unmarshal(b, &renderedWorkflow); err != nil {
+		return nil, errors.Wrap(err, "while unmarshaling to spec")
+	}
+	return &renderedWorkflow.Spec, nil
+}
```

Comment:
I have some concerns regarding naming:
- as suggested previously, this function doesn't create workflow. It decodes Implementation Action arguments into Argo workflow.
- the variable which is currently named `renderdWorkflow` has nothing to do with rendered workflow. It contains Implementation Action arguments with Argo workflow decoded into a proper Argo Workflow type.

So, my suggestion would be to rename it to:

```suggestion
func decodeImplArgsToArgoWorkflow(implArgs map[string]interface{}) (*argo.Workflow, error) {
	var decodedImplArgs = struct {
		Workflow argo.Workflow `json:"workflow"`
	}{}

	b, err := json.Marshal(implArgs)
	if err != nil {
		return nil, errors.Wrap(err, "while marshaling Implementation arguments")
	}
	if err := json.Unmarshal(b, &decodedImplArgs); err != nil {
		return nil, errors.Wrap(err, "while unmarshalling Implementation arguments to Argo Workflow")
	}
	return &decodedImplArgs.Workflow, nil
}
```

---

BTW I wanted to suggest using `mapstructure`, as this is already our dependency, but unfortunately I couldn't make it working for all nested structures with such code:

```go
func decodeImplArgsToArgoWorkflow(implArgs map[string]interface{}) (*argo.Workflow, error) {
	var argoImplArgs struct {
		Workflow argo.Workflow `json:"workflow"`
	}

	decoder, err := mapstructure.NewDecoder(&mapstructure.DecoderConfig{
		Result:  &argoImplArgs,
		TagName: "json",
	})
	if err != nil {
		return nil, errors.Wrap(err, "while creating a decoder for Implementation arguments")
	}

	err = decoder.Decode(implArgs)
	if err != nil {
		  return nil, errors.Wrap(err, "while decoding Implementation arguments to Argo Workflow")
	}
	
	return &argoImplArgs.Workflow, nil
}
```